pkg/types/vsphere/validation/platform.go

```diff
@@ -38,6 +40,24 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
 	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
 		allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+		if network != nil {
+			apiVIPOverlap := false
+			ingressVIPOverlap := false
+			for _, machine := range network.MachineNetwork {
+				if machine.CIDR.Contains(net.ParseIP(p.APIVIP)) {
+					apiVIPOverlap = true
+				}
+				if machine.CIDR.Contains(net.ParseIP(p.IngressVIP)) {
+					ingressVIPOverlap = true
+				}
+			}
+			if !apiVIPOverlap {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, "must overlap with the machine CIDRs provided"))
```

Comment:
Overlap is not the right term. Overlap is used for sets. What you want to say is that the machine CIDR must contain the VIP.
```suggestion
				allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, "must be contained within one of the machine networks"))
```